crates/uopool/src/uopool.rs

```diff
@@ -101,17 +93,34 @@ impl<M: Middleware + 'static> UoPool<M> {
         self.reputation.clear();
     }
 
+    pub async fn validate_user_operation(
+        &self,
+        uo: &UserOperation,
+    ) -> Result<UserOperationValidationOutcome, ValidationError> {
+        self.validator
+            .validate_user_operation(
+                uo,
+                &self.mempool,
+                &self.reputation,
+                UserOperationValidatorMode::Sanity
+                    | UserOperationValidatorMode::Simulation
+                    | UserOperationValidatorMode::SimulationTrace,
+            )
+            .await
+    }
+
+    /// Adds a single validated user operation into the pool
     pub async fn add_user_operation(
         &mut self,
         uo: UserOperation,
-        res: Option<Result<VerificationResult, VerificationError>>,
+        res: Option<Result<UserOperationValidationOutcome, ValidationError>>,
```

Comment:
Related to https://github.com/Vid201/silius/pull/162/files#r1275917583.
Why not take `Option<UserOperationValidationOutcome>`